### PR TITLE
feat: Send case worker ids in the response once CW profiles are created

### PR DESCRIPTION
**Pull request by SujeethKumarAri**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-2207

### Change description ###

Send case worker ids in the response once CW profiles are created

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefCreateTest.java [hunk 1]
```diff
@@ -15,9 +15,9 @@
 import org.springframework.test.context.ActiveProfiles;
 import org.springframework.test.context.TestPropertySource;
 import uk.gov.hmcts.reform.cwrdapi.client.FuncTestRequestHandler;
-import uk.gov.hmcts.reform.cwrdapi.client.response.UserProfileResponse;
 import uk.gov.hmcts.reform.cwrdapi.controllers.request.CaseWorkersProfileCreationRequest;
```
Comment (abhijitdiwan) [inline src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefCreateTest.java:18]: Please use Git hub conventions feat/fix/refactor etc.

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefCreateTest.java [hunk 2]
```diff
@@ -61,6 +62,11 @@ public void whenUserNotExistsInCwrAndSidamAndUp_Ac1() {
         response.then()
                 .assertThat()
                 .statusCode(201);
+
+        CaseWorkerProfileCreationResponse caseWorkerProfileCreationResponse =
+                response.getBody().as(CaseWorkerProfileCreationResponse.class);
+        caseWorkerIds = new ArrayList<>(caseWorkerProfileCreationResponse.getCaseWorkerIds());
```
Comment (abhijitdiwan) [inline src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefCreateTest.java:68]: do we need new array List?

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefCreateTest.java (hunk 1 above)
Comment (SujeethKumarAri) [reply src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefCreateTest.java:18]: I am already using those conventions.
image

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefController.java [hunk 3]
```diff
@@ -145,15 +146,24 @@
...
         }
 
+        CaseWorkerProfileCreationResponse.CaseWorkerProfileCreationResponseBuilder caseWorkerProfileCreationResponse =
+                CaseWorkerProfileCreationResponse
+                        .builder();
         List<CaseWorkerProfile> processedCwProfiles =
                 caseWorkerService.processCaseWorkerProfiles(caseWorkersProfileCreationRequest);
 
         if (!processedCwProfiles.isEmpty()) {
             caseWorkerService.publishCaseWorkerDataToTopic(processedCwProfiles);
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefController.java:156]: CollectionUtils.isNotEmpty. As ! might come sonar Minor

File: src/test/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefControllerTest.java [hunk 4]
```diff
@@ -94,7 +95,7 @@ public void setUp() throws Exception {
                 "lastName","test@gmail.com",1,"userType","region",
                 false,roles,caseWorkerRoleRequests,caseWorkeLocationRequests,caseWorkeAreaRequests);
         caseWorkersProfileCreationRequest.add(cwRequest);
-        cwProfileCreationResponse = new CaseWorkerProfileCreationResponse("");
+        cwProfileCreationResponse = new CaseWorkerProfileCreationResponse("", Collections.emptyList());
```
Comment (abhijitdiwan) [inline src/test/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefControllerTest.java:98]: Can we use builder here?

**Review by abhijitdiwan — COMMENTED** [review]
Please

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefCreateTest.java (hunk 2 above)
Comment (SujeethKumarAri) [reply src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefCreateTest.java:68]: I shall remove and do something like `caseWorkerIds = caseWorkerProfileCreationResponse.getCaseWorkerIds();`

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefCreateTest.java (hunk 1 above)
Comment (SujeethKumarAri) [reply src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefCreateTest.java:18]: done :) - 
image

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefController.java (hunk 3 above)
Comment (SujeethKumarAri) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefController.java:156]: Did not find anything reported by sonar as such - https://sonarcloud.io/project/issues?id=uk.gov.hmcts.reform%3Areform%3A%3Ard-caseworker-ref-api&pullRequest=124&resolved=false :)

File: src/test/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefControllerTest.java (hunk 4 above)
Comment (SujeethKumarAri) [reply src/test/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefControllerTest.java:98]: can do

**Review by abhijitdiwan — APPROVED** [review]

**Review by shreedhar-hmcts — APPROVED** [review]

**Pull request merged**